src/goe/config/orchestration_config.py

```diff
@@ -334,7 +331,6 @@ def __init__(self, do_not_connect=False, **kwargs):
         self.not_null_propagation = (
             self.not_null_propagation.upper() if self.not_null_propagation else None
         )
-        self.num_buckets_max = int(self.num_buckets_max or NUM_BUCKETS_FALLBACK)
         self.num_buckets_threshold = normalise_size_option(
```

Comment:
Is `num_buckets_threshold` still relevant after the removal of synthetic partitioning? It's still present in a number of places.

Reply:
It is still in the code path that automatically sets the hash column for Synapse which is why I left it in place.
To be honest we could probably remove it and only cater for manually supplied hash columns, what do you think?

Reply:
Hmmm, auto-choosing a hash column is a useful feature, but can the `num_buckets_threshold` variable be removed from the logic?

Reply:
I did a bit more digging and this test explains why we use it: https://github.com/gluent/goe/blob/e7a4d4be3b40482d06b28e886ee2d3ced8e8aedb/tests/integration/scenarios/test_offload_hash_column.py#L79

I don't have a problem removing it, we would just need to remove the hash column functionality from Synapse support too.

Reply:
If Synapse requires a hash column then we should auto-select one in the event of a missing option, especially if we want to do a schema offload. That's a lot of configuration to do otherwise.

Reply:
Okay, in which case we should keep the threshold configuration. Current behaviour:
```
Sep 7, 2021 Tables below default bucket threshold size will default to ROUND_ROBIN, else HASH.
HASH keys will be chosen from hash bucket key option or automatically in order of PK, stats and fallback.
```
Without the threshold we'll always have HASH distribution and never use ROUND_ROBIN which is preferable for small tables.

Reply:
Thinking a bit more about this, isn't the number of buckets static in Synapse anyway (60?). Could we modify the threshold calculation?

Reply:
You mean change the default threshold to a higher value?

Reply:
As discussed, as this is a threshold that determines the distribution method, we should rename it to `hash_distribution_threshold`.